File: src/commands/help.js
const { EmbedBuilder } = require('discord.js')

module.exports = {
  name: 'help',
  aliases: ['h', 'commands'],
  run: (bot, message, args) => {
    const embed = new EmbedBuilder()
      .setColor('#B3E04D') // #C3D97B #A6BC64 #EAF198
      .setTitle('Akyuu bot commands\nprefix: **!**')
      .addFields(
        { name: 'to get the data for a move type **![character] [move]**', value: '[move] is the move you want in numpad notation. you can use !wiki controls for more information on that.\nex: !reimu j5a' },
        { name: "if you're new to AOCF, use **!starthere**", value: 'this will get you the links to SogofGod1998 starterpack complete with everything you need to play.' },
        { name: 'wiki', value: 'you can also use !wiki [option] to get to a spacific wiki page or just !wiki by itselft to go to the homepage.\nwiki options: any of the characters, controls, mechanics', inline: true },
        { name: 'wiki arguments', value: '[character]\ncontrols\nmechanics', inline: true },
        { name: 'multiplayer', value: 'win --you won against the other player\nloss --you lost against the other player\nprofile [user] --view your, or someone else\'s, profile' },
        { name: 'other commands', value: 'version  --names the version\nlinks  --gives many useful links\nrepo  --the github repository of Akyuu' }
      )

    message.reply({ embeds: [embed] })
  }
}

File: src/commands/help.rs
use poise::{serenity_prelude::CreateEmbed, CreateReply};

use crate::{Context, Error};

/// View help
#[poise::command(prefix_command, slash_command)]
pub async fn help(
	ctx: Context<'_>,
	//TODO: make it so that /help [command] gives the command description
	// #[description = "Query the help of a specific command"] command: Option<String>,
) -> Result<(), Error> {
	#[rustfmt::skip]
	let embed = CreateEmbed::new().colour(0xB3E04D).title("Akyuu commands").fields([
		(
			"Start here",
			"If you're new to AoCF call `/starthere` to see the links to \
			the starter pack and an explanation on the next steps!",
			false
		),
		(
			"Character moves",
			"Call `/query [character] [input]` to query a moves framedata.",
			false
		),
		(
			"Wiki",
			"Call `/wiki` by itself or with an argument to the right to receive a \
			link to a wiki page.",
			true
		),
		(
			"Wiki arguments",
			"Wiki arguments include any of the characters, `controls`, \
			`mechanics`, `glossary`, and `modding`.",
			true
		),
		(
			"Multiplayer",
			"Call `/win [opponent]`, `/loss [opponent]`, or `/set [wins] [opponent] \
			[losses]` to record your match history. You can also call `/profile` \
			by itself or with a user will show your, or someone else's, records.",
			false
		),
		(
			"Miscellaneous",
			"Other commands include: `/help`, `/links`, and `/repository`.",
			false
		),
	]);

	ctx.send(CreateReply::default().embed(embed).ephemeral(true)).await?;

	Ok(())
}
